Mathlib/RingTheory/Ideal/Pointwise.lean

```diff
@@ -0,0 +1,115 @@
+/-
+Copyright (c) 2024 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import Mathlib.RingTheory.Ideal.Maps
+
+
+/-! # Pointwise instances on `Ideal`s
+
+This file provides the action `Ideal.pointwiseMulAction` which morally matches the action of
+`mulActionSet` (though here an extra `Ideal.span` is inserted).
+
+This actions is available in the `Pointwise` locale.
+
+## Implementation notes
+
+This file is similar (but not identical) to `RingTheory/Subring/Pointwise.lean`.
+Where possible, try to keep them in sync.
+
+-/
+
+
+open Set
+
+variable {M R : Type*}
+
+namespace Ideal
+
+section Monoid
+
+variable [Monoid M] [CommRing R] [MulSemiringAction M R]
+
+/-- The action on an ideal corresponding to applying the action to every element.
+
+This is available as an instance in the `Pointwise` locale. -/
+protected def pointwiseMulSemiringAction : MulSemiringAction M (Ideal R) where
+  smul a I := Ideal.map (MulSemiringAction.toRingHom _ _ a) I
```

Comment:
```suggestion
  smul a I := I.map (MulSemiringAction.toRingHom _ _ a)
```
is there a reason you went for the more longwinded spelling?

Reply:
Furthermore, using comap on a^{-1} for the definition works much better definitionally in the case where M is a group. I'm just moving from my own implementation to this and noticing this annoyance. Monoids can act by endomorphisms; groups have to act via automorphisms. We have to sacrifice nice definitional properties just because we want some generalization to monoids which in practice mathematicians may never use? :-(

Reply:
> Furthermore, using comap on a^{-1} for the definition works much better definitionally in the case where M is a group.

The same could be said for pretty much all the pointwise actions - I took the map approach here mainly for consistency.

> We have to sacrifice nice definitional properties just because we want some generalization to monoids which in practice mathematicians may never use? :-(

This already happened with the definition of `Disjoint`, and surprisingly I haven't really noticed anyone complaining...

Reply:
> just because we want some generalization to monoids which in practice mathematicians may never use?

Maybe `MulSemiringAction` is to blame here; are there any examples of such actions which are a monoid action and not a group action?

Reply:
Probably the endomorphisms of a commutative ring act on the ideals of the ring and I could imagine that someone might need that.